### PR TITLE
ncm-network: device name regex: allow function and device in PCI name

### LINKED ISSUE #1069: ncm-network: /dev/enp4s0f0 and /dev/enp4s0f1 cannot be configured
The regular expressions in network.pm (version 17.2) does not allow to configure /dev/enp4s0f0 and /dev/enp4s0f1. (This naming convention is used in SL7 for "Emulex Corporation OneConnect 10Gb NIC", in SL6 it was eth*) 

Chaning line 280 from
```
    my $dev_regexp='-((eth|seth|em|bond|br|ovirtmgmt|vlan|usb|ib|p\d+p|en(o|(p\d+)?s))\d+|enx[[:xdigit:]]{12})(\.\d+)?';
```
to
```
    my $dev_regexp='-((eth|seth|em|bond|br|ovirtmgmt|vlan|usb|ib|p\d+p|en(o|(p\d+)?s))\d+|enx[[:xdigit:]]{12})(.\d+)?';
```
fixes my problem.

Without this change the name is truncated (leading to enp4s0) and configuration fails.

### PR DESCRIPTION
**Pull request by stdweird**
Fixes #1069

### CODE REVIEW

**Comment by stdweird** [comment]
Unittests and cleanup in #1070 (https://github.com/quattor/configuration-modules-core/pull/1070/commits/aa6ca2ed6cc1220a151fbf39a7e6803c45a772f7 and https://github.com/quattor/configuration-modules-core/pull/1070/commits/7857dc7c874c4fab7c253d37934b9c17ca7e9601)

**Comment by stdweird** [comment]
@ned21 this is the last ncm-network one

**Comment by stdweird** [comment]
retest this please

File: ncm-network/src/main/perl/network.pm [hunk 1]
```diff
@@ -265,7 +265,7 @@ sub Configure
     opendir(DIR, $dir_pref);
     # here's the reason why it only verifies eth, bond, bridge, usb and vlan
     # devices. add regexp at will
-    my $dev_regexp='-((eth|seth|em|bond|br|ovirtmgmt|vlan|usb|ib|p\d+p|en(o|(p\d+)?s))\d+|enx[[:xdigit:]]{12})(\.\d+)?';
+    my $dev_regexp='-((eth|seth|em|bond|br|ovirtmgmt|vlan|usb|ib|p\d+p|en(o|(p\d+)?s(?:\d+f)?(?:\d+d)?))\d+|enx[[:xdigit:]]{12})(\.\d+)?';
```
Comment (ned21) [inline ncm-network/src/main/perl/network.pm]: The comment above this is now extremely out of date. I am guessing the only correct part left is "add regexp at will". Leaving aside that the regexp is also now at the level of expert, what value is actually adding here, aside from needing to update the code whenever new device types come along? Can we simplify this in anyway?

Comment (stdweird) [inline ncm-network/src/main/perl/network.pm]: @ned21 i'd rather not touch this any more than required. please make remarks in the rewrite, eg https://github.com/quattor/configuration-modules-core/pull/1070/commits/d31aca263e18a64a9544348d21abf9a3f7f07f93

**Pull request merged**